src/main/scala/is/hail/expr/Type.scala

```diff
@@ -35,9 +35,9 @@ object Type {
         genScalar,
         genScalar,
         genScalar,
-        genArb.resize(size - 1).map(TArray),
-        genArb.resize(size - 1).map(TSet),
-        Gen.zip(genArb, genArb).map { case (k, v) => TDict(k, v) },
+        Gen.zip(genArb.resize(size - 1), Gen.coin(0.2)).map { case (t, req) => TArray(if (req) !t else t)},
```

Comment:
I think it's easier to parse this if you add a:

```scala
val genNonRequired: Gen[Type] = for {
  t <- gen.resize(size -1)
  req <- Gen.coin(0.2)
} yield if (req) !t else t
```
and use it here:
```scala
genNonRequired.map(TArray(_)),
genNonRequired.map(TSet(_)),
Gen.zip(genArb, genNonRequired).map { case (k, v) => TDict(k, v) }
```

Reply:
genMaybeRequired?

Reply:
bleh, we need a succinct word for "maybe required".

I wonder if we could rework all the names so that no modifier means maybe required, and then we have `genRequired` and `genNotRequired` (or `genNullable` or `genOptional`?)